src/core/mac/sub_mac.cpp

```diff
@@ -286,16 +286,16 @@ void SubMac::BeginTransmit(void)
     error = Get<Radio>().Receive(mTransmitFrame.GetChannel());
     assert(error == OT_ERROR_NONE);
 
-    error = Get<Radio>().Transmit(mTransmitFrame);
-    assert(error == OT_ERROR_NONE);
```

Comment:
If this must be successful, shall we make it no return?

Reply:
`otPlatRadioTransmit()` can/should fail if the allowed radio state transitions are not followed. 
We follow the state transition in sub-mac/mac layer, so expect the call to be successful (therefor the `assert()` here). This is mainly intended to catch any platform implementation issues. I think we should keep it the same to catch/protect for this.